src/bp/core/services/dialog/flow/service.ts

```diff
@@ -126,12 +127,25 @@ export class FlowService {
     this.saveAll(botId, [flow])
   }
 
-  private prepareSaveFlow(flow) {
+  private async _prepareSaveFlow(botId: string, flow: FlowView): Promise<any> {
     const schemaError = validateFlowSchema(flow)
     if (schemaError) {
       throw new Error(schemaError)
     }
 
+    // Find and replace any node name that has changed though all the files that referenced the old name.
+    // FIXME Refactor: Use the node ID instead of the node name to reference it.
+    if (flow.location) {
+      const existingFlow = await this.ghost.forBot(botId).readFileAsObject<FlowView>(FLOW_DIR, flow.location)
+      for (const n1 of existingFlow.nodes) {
```

Comment:
this seems to be with a very high complexity given a big flow, in your replaceInFolders seems like you have a double loop as well ... is there a way to make this simpler ?